cartoframes/viz/kuviz.py

```diff
@@ -45,3 +52,50 @@ def _validate_carto_kuviz(carto_kuviz):
 def _create_carto_kuviz(context, html, name, password=None):
     km = KuvizManager(context.auth_client)
     return km.create(html=html, name=name, password=password)
+
+
+class KuvizPublisher(object):
+    def __init__(self, vmap, context=None):
+        self._layers = deepcopy(vmap.layers)
+        self._context = context
+
+    def set_context(self, context=None):
+        from cartoframes.auth import _default_context
+        self._context = context or _default_context
+
+    def publish(self, html, name, password=None):
+        return Kuviz.create(context=self._context, html=html, name=name, password=password)
+
+    def is_sync(self):
+        syncs = [layer.source.dataset.get_is_saved_in_carto() for layer in self._layers]
+        return False not in syncs
```

Comment:
This is probably a pretty minor performance gain, but turning `syncs` into a tuple/generator is a bit better in theory:
```python
return all(layer.source.dataset.get_is_saved_in_carto() for layer in self._layers)
```

Ref: https://twitter.com/raymondh/status/1125487457443532800

Reply:
:+1: